src/main/java/com/rackspace/salus/event/manage/services/TickScriptBuilder.java

```diff
@@ -61,22 +63,38 @@ public String build(String tenantId, String measurement,  TaskParameters taskPar
     boolean labelsAvailable = false;
     if(taskParameters.getLabelSelector() != null)
       labelsAvailable = true;
+    levelExpression = taskParameters.getInfo();
     return taskTemplate.execute(TaskContext.builder()
         .labels(taskParameters.getLabelSelector() != null ? taskParameters.getLabelSelector().entrySet() : null)
-        .alertId(taskIdGenerator.generateAlertId(tenantId, measurement, taskParameters.getField()))
+        .alertId(taskIdGenerator.generateAlertId(tenantId, measurement, taskParameters.getCritical().getExpression().getField()))
         .labelsAvailable(labelsAvailable)
         .measurement(measurement)
         .details("task={{.TaskName}}")
-        .critExpression(String.format("\"%s\" %s %s", taskParameters.getField(), taskParameters.getComparator(), taskParameters.getThreshold()))
+        .critExpression(String.format("\"%s\" %s %s",
+            taskParameters.getCritical().getExpression().getField(),
+            taskParameters.getCritical().getExpression().getComparator(),
+            taskParameters.getCritical().getExpression().getThreshold()))
+        .infoExpression(taskParameters.getInfo() != null ?
+            String.format("\"%s\" %s %s",
+            taskParameters.getInfo().getExpression().getField(),
+            taskParameters.getInfo().getExpression().getComparator(),
+            taskParameters.getInfo().getExpression().getThreshold()) :
+            null)
+        .warnExpression(taskParameters.getWarning() != null ?
+            String.format("\"%s\" %s %s",
+                taskParameters.getWarning().getExpression().getField(),
+                taskParameters.getWarning().getExpression().getComparator(),
+                taskParameters.getWarning().getExpression().getThreshold()) :
+            null)
         .infoCount(
-            taskParameters.getConsecutiveInfo() != null ?
-                String.format("\"state_count\" >= %d",taskParameters.getConsecutiveInfo())
+            taskParameters.getInfo() != null ?
+                String.format("\"info_count\" >= %d", taskParameters.getInfo().getConsecutiveCount())
                 : null)
         .warnCount(
-          taskParameters.getConsecutiveWarning() != null ?
-            String.format("\"state_count\" >= %d",taskParameters.getConsecutiveWarning())
+          taskParameters.getWarning() != null ?
+            String.format("\"warn_count\" >= %d",taskParameters.getWarning().getConsecutiveCount())
             : null)
-        .critCount(String.format("\"state_count\" >= %d", taskParameters.getConsecutiveCritical()))
+        .critCount(String.format("\"state_count\" >= %d", taskParameters.getCritical().getConsecutiveCount()))
```

Comment:
If you add an `.as('crit_count')` in the template you can be consistent with the expression structure here...and then probably refactor the three of these with a helper method.

Reply:
You got the template change but missed:

```suggestion
        .critCount(String.format("\"crit_count\" >= %d", taskParameters.getCritical().getConsecutiveCount()))
```